### ISSUE TITLE
[Bug]: When updating Providers, the network does not go through the Proxy node in use and Proxy-providers url binding mode not working

### ISSUE THREAD

**Issue by Kingsmanvn-Official** [Fndroid/clash_for_windows_pkg#3607]
### 请认真检查以下清单中的每一项

- [X] 已经搜索过，没有发现类似issue
- [X] 已经搜索过文档，没有发现相关内容
- [X] 已经尝试使用过最新版，问题依旧存在
- [X] 使用的是官方版本（未替换及修改过安装目录程序文件）

### 软件版本

0.20.6

### 操作系统

Windows x64

### 系统版本

Windows 11 Pro version 22H2 (OS Build 22621.674)

### 问题描述

When using Proxy-providers and using url get mode it will fail and return an error.
Now when using the Provider container configuration and updating the Providers, the network goes directly without going through the proxy being run by the profile in the application. (note Proxy-providers mode as I said at the beginning it fails to link the url to get the configuration in any case).
For the rest, I want the author to be able to allow the Proxy-providers and Rule-providers providers to be able to update through the proxy the configuration is using in it.

### 复现步骤

Step 1: Turn on clash vpn (Tun mode)
Step 2: select Providers
Step 3: select update all Providers

### Configure File

```yml
port: 7890
socks-port: 7891
allow-lan: true
bind-address: '*'
mode: rule
ipv6: true
log-level: debug
external-controller: '127.0.0.1:9090'
clash-for-android:
  append-system-dns: true
profile:
  store-selected: false
  store-fake-ip: true
experimental:
  ignore-resolve-fail: true
  sniff-tls-sni: false
dns:
  enable: true
  ipv6: true
  default-nameserver:
    - 192.168.1.1
    - 1.1.1.1
    - 1.0.0.1
    - 8.8.8.8
    - 8.8.4.4
    - 9.9.9.9
    - 119.29.29.29
  enhanced-mode: fake-ip
  fake-ip-range: 198.18.0.1/16
  use-hosts: true
  nameserver:
    - dhcp://system
    - https://dns.google/dns-query
    - https://dns.adguard-dns.com/dns-query
    - https://dns.quad9.net/dns-query
    - https://dns.nextdns.io/fd88ba
    - https://cloudflare-dns.com/dns-query
  fallback:
    - dhcp://system
    - https://dns.google/dns-query
    - https://dns.adguard-dns.com/dns-query
    - https://dns.quad9.net/dns-query
    - https://dns.nextdns.io/fd88ba
    - https://cloudflare-dns.com/dns-query
tun:
  enable: true
  stack: gvisor # (gvisor|system)
  dns-hijack:
    - 198.18.0.2:53
  auto-route: true
  auto-detect-interface: true
proxies:
  - name: '🇸🇬 SG snell'
    type: snell
    server: xxx
    port: xxx
    psk: xxx
    version: 3
    udp: true
    obfs-opts:
      mode: tls
      host: xxx
proxy-groups:
    - name: KINGSMANVN
      type: select
      use:
        - yuna
      proxies:
        - "🇸🇬 SG snell"
proxy-providers:
  yuna:
    type: http
    url: "https://xxx"
    path: ./profiles/yuna.yml
    interval: 3600
    health-check:
      enable: true
      interval: 420
      url: http://www.gstatic.com/generate_204
rule-providers:
  hostsvn:
    behavior: domain # domain, ipcidr or classical (premium core only)
    type: http
    url: "https://xxx/hostsVN-clash-rule.yaml"
    interval: 72240
    path: ./ruleset/hostsvn.yaml
  reject:
    type: http
    behavior: domain
    url: "https://xxx/reject.txt"
    path: ./ruleset/reject.yaml
    interval: 86400
rules:
  - RULE-SET,hostsvn,REJECT
  - RULE-SET,reject,REJECT
  - MATCH,KINGSMANVN
  
```

### 日志文件

<details><summary>logs</summary><p>

```
进入Home Directory/logs文件夹，找到日志文件，将文件拖动至输入框或点击界面左下角小字，将打开文件内容粘贴替换这行
```

</p></details>

In the log there is no error related to the error I am describing

### 其他补充

Config use Proxy-providers mode get from url link returns error on enable failed to run and automatically exits selected profile to use:
![image](https://user-images.githubusercontent.com/94727017/197762319-98cc5956-8091-4c09-a8dc-f7000a28edf9.png)
![image](https://user-images.githubusercontent.com/94727017/197813827-183f4e83-f564-4f26-97ab-fef238d282c4.png)

Now when using the Provider container configuration and updating the Providers, the network goes directly without going through the proxy being run by the in-app profile:
![image](https://user-images.githubusercontent.com/94727017/197761298-86c50fe3-217d-4f0c-90b7-2f0c531dbe42.png)

**Comment by yi-Xu-0100** [comment]
show the tun config or add more nameserver in dns profiles

**Comment by yi-Xu-0100** [comment]
The provider can not use the proxy under clash itself, it is a core issue: <https://github.com/Dreamacro/clash/issues>

The error says: all DNS requests failed, first error: interface not found.

You may need to disable `auto-route` and set the `interface-name` in your profiles.

**Comment by Kingsmanvn-Official** [comment]
@yi-Xu-0100 It's still error, nothing has changed bro
![image](https://user-images.githubusercontent.com/94727017/197814019-22d15abd-a530-4299-9b7b-04a0b03e3bc2.png)

**Comment by yi-Xu-0100** [comment]
@Kingsmanvn-Official 

It is a YAML syntax issue. You need to check the file in your local. 👀

**Comment by Kingsmanvn-Official** [comment]
> @Kingsmanvn-Official
> 
> It is a YAML syntax issue. You need to check the file in your local. 👀

@yi-Xu-0100 The url link for use in Proxy-providers that's the complete configuration I've made separately into a profile for perfectly normal use with no syntax errors. I used the same file in my local to push into the Proxy-providers section of the main config I use because the url fetching mode was broken. And when I update the Rule-providers to get the url link, the provider can't use the proxy under clash itself.
